src/electron/process.ts

```diff
@@ -22,89 +23,107 @@ import * as path from 'path';
 //       (which may be immediately after calling #startInternal if, e.g. the binary cannot be
 //       found).
 export class ChildProcessHelper {
-  private process?: ChildProcess;
-  protected isInDebugMode = false;
+  private readonly processName: string;
+  private subProcess?: ChildProcess = null;
+  private exitCodePromise?: Promise<number | string> = Promise.resolve('not started');
+  private isDebug = false;
 
-  private exitListener?: (code?: number, signal?: string) => void;
   private stdErrListener?: (data?: string | Buffer) => void;
 
-  constructor(private path: string) {}
+  constructor(private readonly path: string) {
+    this.processName = basename(this.path);
+  }
 
   /**
    * Starts the process with the given args. If enableDebug() has been called, then the process is
-   * started in verbose mode if supported.
+   * started in verbose mode if supported. This method will only start the process, it will not
+   * wait for it to be ended. Please use stop() or waitForExit() instead.
    * @param args The args for the process
    */
-  launch(args: string[]) {
-    this.process = spawn(this.path, args);
-    const processName = path.basename(this.path);
-
-    const onExit = (code?: number, signal?: string) => {
-      if (this.process) {
-        this.process.removeAllListeners();
-      }
-      if (this.exitListener) {
-        this.exitListener(code, signal);
-      }
-
-      logExit(processName, code, signal);
-    };
-    const onStdErr = (data?: string | Buffer) => {
-      if (this.isInDebugMode) {
-        console.error(`[STDERR - ${processName}]: ${data}`);
-      }
-      if (this.stdErrListener) {
-        this.stdErrListener(data);
-      }
-    };
-    this.process.stderr.on('data', onStdErr.bind(this));
-
-    if (this.isInDebugMode) {
-      // Redirect subprocess output while bypassing the Node console.  This makes sure we don't
-      // send web traffic information to Sentry.
-      this.process.stdout.pipe(process.stdout);
-      this.process.stderr.pipe(process.stderr);
+  launch(args: string[]): void {
+    if (this.subProcess) {
+      throw new Error(`subprocess ${this.processName} has already been launched`);
     }
+    this.subProcess = spawn(this.path, args);
+    this.exitCodePromise = new Promise(resolve => {
+      const onExit = (code?: number, signal?: string) => {
+        if (this.subProcess) {
+          this.subProcess.removeAllListeners();
+          this.subProcess = null;
+        } else {
+          // When listening to both the 'exit' and 'error' events, guard against accidentally
+          // invoking handler functions multiple times.
+          return;
+        }
+
+        logExit(this.processName, code, signal);
+        resolve(code ?? signal);
```

Comment:
I'm not a big fan of returning a value with multiple types.
Can you make it some Error object instead (maybe we need something like ExitError(number) and SignalError(string)?

Also, what do you think about calling `launch()` `run()`, and make it return a `Promise<void>` that resolves on successful exit and rejects on error?

I think it's easier to reason and removes the need for `waitForEnd()`, as you can just await the Promise.

Reply:
I think that makes sense, we can let the caller decide what to do with the returned `Promise` of `launch()`.